addons/mail/models/mail_activity.py

```diff
@@ -166,47 +169,75 @@ def _check_access(self, operation):
         for model, res_ids in activity_to_documents.items():
             self.env[model].check_access_rights(doc_operation, raise_exception=True)
             try:
                 self.env[model].browse(res_ids).check_access_rule(doc_operation)
             except exceptions.AccessError:
-                raise exceptions.AccessError(
-                    _('The requested operation cannot be completed due to security restrictions. Please contact your system administrator.\n\n(Document type: %s, Operation: %s)') %
-                    (self._description, operation))
+                self._raise_access_exception(operation)
 
     @api.model
     def create(self, values):
         # already compute default values to be sure those are computed using the current user
         values_w_defaults = self.default_get(self._fields.keys())
         values_w_defaults.update(values)
 
+        # Reminder have no summary (for display name) It will add first line of note to summary.
+        if 'res_model_id' not in values and 'summary' not in values:
+            values_w_defaults['summary'] = self._summary_from_note(values.get('note', _('Reminder')))
+
         # continue as sudo because activities are somewhat protected
         activity = super(MailActivity, self.sudo()).create(values_w_defaults)
         activity_user = activity.sudo(self.env.user)
         activity_user._check_access('create')
-        self.env[activity_user.res_model].browse(activity_user.res_id).message_subscribe(partner_ids=[activity_user.user_id.partner_id.id])
-        if activity.date_deadline <= fields.Date.today():
-            self.env['bus.bus'].sendone(
-                (self._cr.dbname, 'res.partner', activity.user_id.partner_id.id),
-                {'type': 'activity_updated', 'activity_created': True})
+        if activity.res_id and activity.res_model:
+            self.env[activity_user.res_model].browse(activity_user.res_id).message_subscribe(partner_ids=[activity_user.user_id.partner_id.id])
+            if activity.date_deadline <= fields.Date.today():
+                self.env['bus.bus'].sendone(
+                    (self._cr.dbname, 'res.partner', activity.user_id.partner_id.id),
+                    {'type': 'activity_updated', 'activity_created': True})
         return activity_user
 
     @api.multi
     def write(self, values):
-        self._check_access('write')
+        self._check_access('write',values)
+        if not self.env.user._is_admin() and ('res_model' in values or 'res_id' in values or 'res_model_id' in values):
+            self._raise_access_exception('write',_("You cannot change a reminder's parent record."))
         if values.get('user_id'):
             pre_responsibles = self.mapped('user_id.partner_id')
-        res = super(MailActivity, self.sudo()).write(values)
 
+        activities = self.env['mail.activity']
+        reminders = self.env['mail.activity']
```

Comment:
Why not
```
reminders = self.filtered(lambda a: not a.res_id and not a.res_model) if values.get('note') and 'summary' not in values else self.env['mail.activity']
activities = self - reminders
if reminders:
  update values
  super(Activity, reminders).write().
res = super(Activity, activities).write()...
```

Reply:
-The value should be updated only after super call is made for activities
-I think it's better to combine the result for all call of super.write(), no?